src/components/ec-modal/ec-modal.spec.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable no-use-before-define */
+import { mount } from '@vue/test-utils';
+import EcModal from './ec-modal.vue';
+
+describe('EcModal', () => {
+  it('should not render the modal', () => {
+    const wrapper = mount(EcModal);
+    expect(wrapper.find('.ec-modal').exists()).toBe(false);
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should render basic modal', () => {
+    const wrapper = mount(EcModal, {
+      propsData: {
+        showModal: true,
+      },
+      slots: {
+        main: '<p>Before we can process your application we need you to upload your management accounts. You can do this now or leave it for later.</p>',
+      },
+    });
+    expect(wrapper.element).toMatchSnapshot();
+  });
+  it('should render close button', () => {
+    const wrapper = mount(EcModal, {
+      propsData: {
+        showModal: true,
+      },
+    });
+    expect(wrapper.find('.ec-modal__close').exists()).toBe(true);
+    expect(wrapper.element).toMatchSnapshot();
```

Comment:
BTW, to make the snapshots smaller, you can take a snapshots of partial html, e.g. in this case:
```
expect(wrapper.find('.ec-modal__header').element).toMatchSnapshot();
```
... if you know the close button is in the header. See also: https://kentcdodds.com/blog/effective-snapshot-testing/

There are many places in this file where it can be used, e.g. testing if the buttons are present in footer.

Reply:
Awesome cheers :)